Assets/UI/WorldView/plotinfo.lua

```diff
@@ -188,46 +188,37 @@ function ShowPurchases()
-            else
-              pInstance.PurchaseButton:GetTextControl():SetColorByName("ResGoldLabelCS");
-            end
-            pInstance.PurchaseButton:RegisterCallback( Mouse.eLClick, function() OnClickPurchasePlot( index ); end );
-            pInstance.PurchaseAnim:SetColor( (goldCost > playerGold ) and 0xbb808080 or 0xffffffff ) ;
-            pInstance.PurchaseAnim:RegisterEndCallback( OnSpinningCoinAnimDone );
-            if (goldCost > playerGold ) then
-              pInstance.PurchaseButton:ClearMouseEnterCallback();
-              pInstance.PurchaseButton:SetToolTipString( Locale.Lookup("LOC_PLOTINFO_YOU_NEED_MORE_GOLD_TO_PURCHASE", goldCost - math.floor(playerGold) ));
-            else
-              pInstance.PurchaseButton:RegisterMouseEnterCallback( function() OnSpinningCoinAnimMouseEnter(pInstance.PurchaseAnim); end );
-              pInstance.PurchaseButton:SetToolTipString("");
-            end
-            if (index == pNextPlotID ) then
-              CQUI_SetupNextPlotButton(pInstance, pCityCulture, TurnsUntilExpansion);
-            end
-            pInstance.PurchaseButton:SetHide( false );
-            table.insert( m_uiPurchase, pInstance );
+        local index:number = kPlot:GetIndex();
+        local pInstance:table = GetInstanceAt( index );
+        if pInstance ~= nil then
+          local goldCost = cityGold:GetPlotPurchaseCost( index );
+          pInstance.PurchaseButton:SetText(tostring(goldCost));
+          AutoSizeGridButton(pInstance.PurchaseButton,51,30,25,"H");
+          pInstance.PurchaseButton:SetDisabled( goldCost > playerGold );
+          if( goldCost > playerGold) then
+            pInstance.PurchaseButton:GetTextControl():SetColorByName("TopBarValueCS");
+          else
+            pInstance.PurchaseButton:GetTextControl():SetColorByName("ResGoldLabelCS");
+          end
+          pInstance.PurchaseButton:RegisterCallback( Mouse.eLClick, function() OnClickPurchasePlot( index ); end );
+          pInstance.PurchaseAnim:SetColor( (goldCost > playerGold ) and 0xbb808080 or 0xffffffff ) ;
+          pInstance.PurchaseAnim:RegisterEndCallback( OnSpinningCoinAnimDone );
+          if (goldCost > playerGold ) then
+            pInstance.PurchaseButton:ClearMouseEnterCallback();
+            pInstance.PurchaseButton:SetToolTipString( Locale.Lookup("LOC_PLOTINFO_YOU_NEED_MORE_GOLD_TO_PURCHASE", goldCost - math.floor(playerGold) ));
           else
-            UI.DataError("Failed to get instance for plot purchase button with index #"..tostring(kPlot:GetIndex()));
+            pInstance.PurchaseButton:RegisterMouseEnterCallback( function() OnSpinningCoinAnimMouseEnter(pInstance.PurchaseAnim); end );
+            pInstance.PurchaseButton:SetToolTipString("");
           end
-          table.insert(m_kLensMask[KEY_PLOT_PURCHASE], plotId);
+          if (index == pNextPlotID ) then
+            CQUI_SetupNextPlotButton(pInstance, pCityCulture, TurnsUntilExpansion);
```

Comment:
We need to remove the CQUI_SetupNextPlotButton function and all the call to it (line 227 and 214) since I added the vanilla way to show the growth tile now.